R/scale-manual.r

```diff
@@ -132,8 +132,8 @@ manual_scale <- function(aesthetic, values = NULL, breaks = waiver(), ..., limit
     force(values)
   }
 
-  if (is.null(limits)) {
-    limits <- names(values)
+  if (is.null(limits) && !is.null(names(values))) {
+    limits <- function(x) {intersect(x, names(values))}
```

Comment:
```suggestion
    limits <- function(x) intersect(x, names(values))
```

Tidyverse style guide says:

> `{` should be the last character on the line. 
(https://style.tidyverse.org/syntax.html#indenting)

So, it's probably recommended to write this as either

``` r
    limits <- function(x) intersect(x, names(values))
```

or

``` r
    limits <- function(x) {
      intersect(x, names(values))
    }
```